storage/innobase/handler/ha_innodb.cc

```diff
@@ -12416,6 +12421,43 @@ const char *create_table_info_t::create_options_are_invalid() {
 static const LEX_STRING yes_string = {C_STRING_WITH_LEN("Y")};
 static const LEX_STRING keyring_string = {C_STRING_WITH_LEN("KEYRING")};
 
+void ha_innobase::adjust_encryption_key_id(HA_CREATE_INFO *create_info,
+                                           dd::Properties *options) noexcept {
+  LEX_STRING *encrypt_type = &create_info->encrypt_type;
+
+  if (false == create_info->was_encryption_key_id_set) {
+    if (Encryption::should_be_keyring_encrypted(encrypt_type->str)) {
+      create_info->encryption_key_id =
+          THDVAR(current_thd, default_encryption_key_id);
+      create_info->was_encryption_key_id_set = true;
+    }
+  } else if (Encryption::is_master_key_encryption(encrypt_type->str) ||
+             Encryption::none_explicitly_specified(encrypt_type->str)) {
+    // if it is encrypted table with Master key encryption or marked as not to
+    // be encrypted and alter table does not have ENCRYPTION_KEY_ID - mark
+    // encryption key id as not set.
+
+    push_warning_printf(current_thd, Sql_condition::SL_WARNING,
+                        HA_WRONG_CREATE_OPTION,
+                        Encryption::none_explicitly_specified(encrypt_type->str)
```

Comment:
This function looks like a question type "is_" ?
But it modifies the table options as well. Please use a better name

Reply:
done